src/herbpy/action/pouring.py

```diff
@@ -0,0 +1,72 @@
+import openravepy, time, prpy, math, numpy
+from prpy.action import ActionMethod
+
+@ActionMethod
+def MoveCupAndPour(robot, table, manip_pitcher, cup, pitcher):
+    """
+    @param robot The robot performing the pouring
+    @param table The table the objects are on
+    @param manip_pitcher The arm that is manipulating the pitcher
+    @param cup The cup to pour into
+    @param pitcher The pitcher to pour
+    """	
+
+    manip_cup = robot.right_arm
+    if manip_pitcher == robot.right_arm:
+        manip_cup = robot.left_arm
+
+    env = robot.GetEnv()
+    with env:
+        pitcher_start = pitcher.GetTransform()
+        pitcher_aabb = pitcher.ComputeAABB()
+        cup_pose = cup.GetTransform()
+        cup_aabb = cup.ComputeAABB()
+        manip_cup_pose = manip_cup.GetEndEffectorTransform()
+        manip_pitcher_pose = manip_pitcher.GetEndEffectorTransform()
+
+    p_tsr, min_tilt, max_tilt = robot.tsrlibrary(pitcher, 'pour')
+    traj = manip_pitcher.PlanToTSR(p_tsr, execute=False)
+    cspec = traj.GetConfigurationSpecification()
+    wpt = traj.GetWaypoint(traj.GetNumWaypoints() - 1)
+    config = cspec.ExtractJointValues(wpt, robot, manip_pitcher.GetArmIndices())
+    with env, robot:
```

Comment:
Please create an explicit `RobotStateSaver` with the flags you care about, instead of using the omnibus `with robot` form. In this case, I believe you only need `LinkTransformations`.